mavis/assemble.py

```diff
@@ -389,11 +390,14 @@ def assemble(
     assembly.trim_noncutting_paths_by_freq(min_edge_trim_weight)
 
     path_scores = {}
+    for component in sorted([i for i in digraph_connected_components(assembly)]):
+
+        # copy here so filtering is done per component not on the full assembly graph
+        component_graph = assembly.copy()
```

Comment:
Did this actually affect anything? components are by definition isolated

Reply:
See comment, the components are isolated, but the trimming that happens in the pull_contigs_from_components affected the entire graph, including downstream coponents.

Reply:
Do we end up with more contigs this way?

Reply:
yes we do end up with more contigs. but they most of them get filtered by the next step

Reply:
Do we still need to copy when we are also sorting the components first? shouldn't that make it deterministic?

Reply:
I think the sorting does make it deterministic, but I thought it might be better if filter each component separately.

Reply:
copying the graph could end up being a pretty heavy performance hit the larger the assembly is. If we don't need to copy the entire graph it will be better not to

Reply:
I was wrong, the sorting doesn't affect the deterministic behavior.

Reply:
fixed using assembly.subgraph instead of assembly.copy